File: python/friendly.py
import sys

def sumDevidors(victim):
    sum = 1
    i = 2
    while i*i < victim:
        if victim % i == 0:
            sum += + i + (victim / i)
        i += 1
    if i*i == victim:
        sum += + i
    return sum

top = int(sys.argv[1])
print "top is " + str(top)
for candidate in range(1, top):
    mate = sumDevidors(candidate)
    if mate <= candidate:
        continue
    if candidate == sumDevidors(mate):
        print candidate, mate

File: js/friendly.js
function sumDevidors(victim) {
	let sum = 1;
	let i;
	for(i = 2; i*i < victim; i++) {
		if (victim % i === 0) {
			sum = sum + i + (victim / i);
		}
	}
	if (i*i === victim) return sum + i;
	return sum;
}

let top = process.argv[2]

for (let candidate = 1; candidate < top; candidate++) {
	let mate = sumDevidors(candidate)
	if(mate <= candidate) continue;
	if(candidate === sumDevidors(mate)) {
		console.log(candidate, mate);
	}
}
